### PR TITLE
Refactor to use Ansible package module

### PR DESCRIPTION
**Pull request by mtnbikenc**
The Ansible package module will call the correct package manager for the underlying OS.

### CODE REVIEW

File: docs/best_practices_guide.adoc [hunk 1]
```diff
@@ -490,11 +490,11 @@ This is almost always more desirable than an empty list, string, etc.
 
 === Yum and DNF
 '''
-[[Package-installation-MUST-use-ansible-action-module-to-abstract-away-dnf-yum]]
+[[Package-installation-MUST-use-ansible-pacakge-module-to-abstract-away-dnf-yum]]
```
Comment (miabbott) [inline docs/best_practices_guide.adoc]: misspelled package

File: docs/best_practices_guide.adoc [hunk 2]
```diff
@@ -533,6 +536,6 @@ modules.
 ---
 # tasks.yml
 - name: Install etcd (for etcdctl)
-  action: "{{ ansible_pkg_mgr }} name=etcd state=latest"
+  pacakge: name=etcd state=latest
```
Comment (miabbott) [inline docs/best_practices_guide.adoc]: another misspell of package

File: docs/best_practices_guide.adoc [hunk 3]
```diff
@@ -507,21 +507,24 @@ This is almost always more desirable than an empty list, string, etc.
 This is done primarily because if you're registering the result of the
 installation and you have two conditional tasks based on whether or not yum or
 dnf are in use you'll end up inadvertently overwriting the value. It also
-reduces duplication. name= and state=present are common between dnf and yum
-modules.
```
Comment (detiber) [inline docs/best_practices_guide.adoc]: Should lines 500-509 be removed as well, since syncing parameters between yum/dnf modules is no longer needed?

**Comment by mtnbikenc** [comment]
Updated, rebased and squashed.

**Comment by sdodson** [comment]
@maxamillion can you look into testing this for fedora installs?

**Comment by sdodson** [comment]
aos-ci-test

**Comment by mtnbikenc** [comment]
Corrected indent issue causing --syntax-check failure.

**Comment by maxamillion** [comment]
@sdodson Yes, I'll try and get to this sometime this afternoon or tomorrow.

**Comment by detiber** [comment]
lgtm pending Fedora tests

**Comment by mtnbikenc** [comment]
@maxamillion Could you comment on Fedora testing status?  Thanks!

**Comment by maxamillion** [comment]
👍

**Comment by detiber** [comment]
aos-ci-test

**Pull request merged**